osu.Game/Screens/SelectV2/Carousel.cs

```diff
@@ -463,27 +471,27 @@ private void refreshAfterSelection()
                 return;
             }
 
-            float spacing = SpacingBetweenPanels;
             int count = carouselItems.Count;
 
             Selection prevKeyboard = currentKeyboardSelection;
 
             // We are performing two important operations here:
-            // - Update all Y positions. After a selection occurs, panels may have changed visibility state and therefore Y positions.
             // - Link selected models to CarouselItems. If a selection changed, this is where we find the relevant CarouselItems for further use.
+            // - Update all Y positions. After a selection occurs, panels may have changed visibility state and therefore Y positions.
+
```

Comment:
This change of order is necessary as the Y position now has a dependency on which beatmap is selected (see `BeatmapCarousel.GetSpacingBetweenItems`). Without that dependency, there's no way to make the expanded/selected beatmap set not overlap with the beatmap set above it, which looks broken to my eyes.

With overlap:
![CleanShot 2025-02-03 at 02 32 52](https://github.com/user-attachments/assets/9544c83e-bc49-4c28-ba3b-eff591881cc7)

Without overlap:
![CleanShot 2025-02-03 at 02 33 06](https://github.com/user-attachments/assets/9d28a489-aa8a-4c60-95c8-03ef594cd2ec)

Reply:
This undoes an important optimisation (you added back a second traversal of every item, which is costly), so will require scrutiny.

Reply:
For now please undo this one. Let's keep this PR as implementing the design and making minimal-to-no changes to the carousel code. You're welcome to force push away any changes if it will help with conflict avoidance with other carousel PRs.